lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -352,6 +352,25 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  return dosomething_reportback_save($values, $user);
+  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+
+  if (! $transaction_id) {
+    $rogue_reportback = dosomething_rogue_send_reportback_to_rogue($values, $user);
+
+    // Make sure the reportback exists (meaning it got back from Rogue).
+    $rbid = dosomething_reportback_exists($rogue_reportback['data']['campaign_id'], $rogue_reportback['data']['campaign_run_id'], $rogue_reportback['data']['drupal_id']);
+
+    // Store reference to the rb in rogue, redirect user to permalink page.
+    if ($rbid) {
```

Comment:
oh yes I see what you mean now, @angaither ! So if there is no transaction ID, then it sends it to Rogue and Rogue sends back to this function and should now have a transaction id. Since it now has a transaction id, it'll move on to the `else` statement and then save.